src/Akka.Hosting/AkkaHostedService.cs

```diff
@@ -33,6 +34,7 @@ public async Task StartAsync(CancellationToken cancellationToken)
             try
             {
                 _actorSystem = _serviceProvider.GetRequiredService<ActorSystem>();
+                _coordinatedShutdown = CoordinatedShutdown.Get(_actorSystem);
```

Comment:
Cache `CoordinatedShutdown` instance so in case `ActorSystem` is terminate first it can't cause the `CoordinatedShutdown.Get` to NRE when resolving the `ActorSystemExtension` internally.